test/e2e/framework/node/wait.go

```diff
@@ -207,30 +207,28 @@ func checkWaitListSchedulableNodes(c clientset.Interface) (*v1.NodeList, error)
 	return nodes, nil
 }
 
-// CheckReadyForTests returns a method usable in polling methods which will check that the nodes are
-// in a testable state based on schedulability.
+// CheckReadyForTests returns a function which will return 'true' once the number of ready nodes is above the allowedNotReadyNodes threshold (i.e. to be used as a global gate for starting the tests).
 func CheckReadyForTests(c clientset.Interface, nonblockingTaints string, allowedNotReadyNodes, largeClusterThreshold int) func() (bool, error) {
 	attempt := 0
-	var notSchedulable []*v1.Node
 	return func() (bool, error) {
 		attempt++
-		notSchedulable = nil
+		var nodesNotReadyYet []*v1.Node
 		opts := metav1.ListOptions{
 			ResourceVersion: "0",
-			FieldSelector:   fields.Set{"spec.unschedulable": "false"}.AsSelector().String(),
+			// remove uncordoned nodes from our calculation, TODO refactor after node v2 API removes that semantic.
```

Comment:
i'm not aware of the Node v2 plans. is this real ATM?
if not possibly we should omit the TODO.

Reply:
spec.Unschedulable i think is fundamentally going to change at some point, right?

Reply:
( i believe this related to taintnodebycondition and so on, but in general, this is known to be a sore point of the API that is supposed to be, eventually, revisited, so i think we should keep the v2 node api comes up which will have a cleaner way of doing this )

Reply:
SGTM, hope the comment is not misleading to readers, as in "Node v2 is in the works as we speak".

Reply:
actually...
> TODO refactor after node v2 API removes that semantic.

we don't know that for sure unless SIG Node confirms it 100%, so i'd avoid putting words in their mouth here.

i'd reword to:
> TODO refactor if Node v2 API removes that semantic.